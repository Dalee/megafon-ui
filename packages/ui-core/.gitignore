.idea
*.iml
node_modules

src/**/*.css
src/index.ts
src/components/**/*.d.ts
dist
styles/base.less
coverage
.docz
docs
.vscode
.DS_Store

.npmrc
