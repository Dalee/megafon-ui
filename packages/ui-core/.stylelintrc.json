{
  "rules": {
    "color-hex-case": "upper",
    "color-no-invalid-hex": true,
    "color-hex-length": "long",
    "font-family-name-quotes": "always-where-recommended",
    "function-calc-no-unspaced-operator": true,
    "function-linear-gradient-no-nonstandard-direction": true,
    "function-comma-space-after": "always",
    "function-comma-space-before": "never",
    "function-parentheses-space-inside": "never",
    "number-leading-zero": "always",
    "number-no-trailing-zeros": true,
    "string-no-newline": true,
    "string-quotes": "single",
    "length-zero-no-unit": true,
    "unit-case": "lower",
    "unit-no-unknown": true,
    "value-keyword-case": "lower",
    "value-no-vendor-prefix": true,
    "property-case": "lower",
    "property-no-unknown": true,
    "keyframe-declaration-no-important": true,
    "declaration-bang-space-after": "never",
    "declaration-block-no-duplicate-properties": true,
    "declaration-block-trailing-semicolon": "always",
    "declaration-block-semicolon-space-before": "never",
    "block-no-empty": true,
    "block-opening-brace-newline-after": "always",
    "block-closing-brace-newline-before": "always",
    "selector-attribute-brackets-space-inside": "never",
    "selector-pseudo-class-no-unknown": true,
    "selector-pseudo-element-no-unknown": true,
    "selector-type-no-unknown": true,
    "selector-max-class": 3,
    "selector-no-qualifying-type": [true, { "ignore": ["attribute"] }],
    "selector-max-id": 0,
    "max-nesting-depth": 5,
    "indentation": 4,
    "declaration-bang-space-before": "always",
    "media-feature-name-no-unknown": true,
    "at-rule-no-unknown": true,
    "comment-no-empty": true,
    "no-descending-specificity": true,
    "no-duplicate-at-import-rules": true,
    "no-duplicate-selectors": true,
    "no-empty-source": true,
    "no-extra-semicolons": true,
    "color-named": "never",
    "declaration-block-single-line-max-declarations": 1,
    "selector-max-empty-lines": 0,
    "function-url-quotes": "always",
    "function-whitespace-after": "always",
    "custom-property-empty-line-before": "never",
    "declaration-empty-line-before": "never",
    "declaration-block-semicolon-newline-after": "always",
    "block-closing-brace-empty-line-before": "never",
    "block-closing-brace-newline-after": "always"
  }
}
