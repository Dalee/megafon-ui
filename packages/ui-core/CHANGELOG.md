# Changelog

## Unreleased
- **Grid**
    - add examples in .mdx

## Release 1.0.5
- **GridColumn**
    - fixed empty array for prop 'wide'

## Release 1.0.4
- **Carousel**
    - fix 'pointer-events' for disabled arrow

## Release 1.0.3
- **Grid**
    - add new component
    - add typescript
- **GridColumn**
    - add new component
    - add typescript

## Release 1.0.2
- **Carousel**
    - fix bottom padding in less

Old CHANGELOG is available [here](https://github.com/MegafonWebLab/megafon-ui/blob/master/CHANGELOG_old.md)
