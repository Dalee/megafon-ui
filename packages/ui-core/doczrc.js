export default {
    title: "MegaFon UI",
    description: "Megafon React UI Kit",
    typescript: true
};
