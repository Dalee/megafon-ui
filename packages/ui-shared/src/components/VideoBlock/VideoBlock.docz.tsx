import { IContent } from './VideoBlock';
import video from '../../../demo/video.mp4';

const content: IContent = {
    title: 'Lorem ipsum',
    description: [
        'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.',
    ],
    href: '#',
    buttonTitle: 'Lorem ipsum',
    onButtonClick: () => alert('button click'),
};

const youtubeVideoId = '2Sps5MnvlKM';

export { content, video, youtubeVideoId };
