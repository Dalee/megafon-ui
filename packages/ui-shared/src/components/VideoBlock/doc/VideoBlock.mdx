---
name: VideoBlock
menu: Конструктор
route: components/video-block
---

import Example from './VideoBlock.example.mdx';
import Props from './VideoBlock.props.mdx';

# VideoBlock

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
