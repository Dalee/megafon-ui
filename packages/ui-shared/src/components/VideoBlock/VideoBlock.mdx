---
name: VideoBlock
menu: Constructor
route: components/video-block
---

import { Playground, Props } from 'docz';
import VideoBlock from './VideoBlock';
import { content, video, youtubeVideoId } from './VideoBlock.docz';

# VideoBlock

## Properties

<Props of={VideoBlock} />

## Included props

```javascript

interface IContent {
    title: string;
    description: string[];
    buttonTitle: string;
    href?: string;
    onButtonClick?: (e: React.SyntheticEvent<EventTarget>) => void;
}

const content: IContent = {
    title: 'Lorem ipsum',
    description: [
        'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.',
        'Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. '
    ],
    href: '#',
    buttonTitle: 'Lorem ipsum',
    onButtonClick: () => alert('button click'),
}

const videoSrc = 'video.mp4' | '2Sps5MnvlKM';
```

## Video without content

<Playground>
    <VideoBlock
        videoSrc={video}
    />
</Playground>

## Video with content

<Playground>
    <VideoBlock
        content={content}
        videoSrc={video}
    />
</Playground>

## Video from Youtube

<Playground>
    <VideoBlock
        videoSrc={youtubeVideoId}
        videoType="youtube"
    />
</Playground>

## Video from Youtube with content

<Playground>
    <VideoBlock
        content={content}
        videoSrc={youtubeVideoId}
        videoType="youtube"
    />
</Playground>
