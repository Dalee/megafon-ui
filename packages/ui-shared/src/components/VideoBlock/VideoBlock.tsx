import * as React from 'react';
import PropTypes from 'prop-types';
import './VideoBlock.less';
import { Header, Button, Paragraph, ContentArea, Grid, GridColumn, cnCreate } from '@megafon/ui-core';

interface IContentData {
    title: string;
    description: string[];
    buttonTitle: string;
    href?: string;
    onButtonClick?: (e: React.SyntheticEvent<EventTarget>) => void;
}

export interface Props {
    /** Content */
    content?: IContentData;
    /** Media type */
    mediaType?: 'video' | 'youtube' | 'gif';
    /** Media source */
    mediaSource: string;
    /** Custom className */
    className?: string;
}

const cn = cnCreate('mfui-video-block');
const VideoBlock: React.FC<Props> = ({
    content,
    mediaType = 'video',
    mediaSource,
    className,
}) => {

    const renderSource = (sourceType): JSX.Element => {
        switch (sourceType) {
            case 'youtube':
                return (
                    <iframe className={cn('youtube-wrapper')}
                            src={mediaSource}
                            frameBorder="0"
                            allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
                            allowFullScreen />
                );
            case 'gif':
                return (
                    <img className={cn('image')} src={mediaSource} alt="gif"/>
                );
            default:
                return (
                    <video className={cn('video')} autoPlay muted>
                        <source className={cn('source')} src={mediaSource} type="video/mp4" />
                    </video>
            );
        }
    };

    const renderContent = (contentData: IContentData): JSX.Element => {
        const { title, description, href, buttonTitle, onButtonClick } = contentData;

        return (
            <div className={cn('content')}>
                <Header as="h3" className={cn('header')}>
                    {title}
                </Header>
                {description && description.map((paragraph, i) => (
                    <Paragraph key={i + paragraph} className={cn('text')} hasMargin={false}>
                        {paragraph}
                    </Paragraph>
                ))}
                <Button className={cn('button')} href={href} onClick={onButtonClick}>
                    {buttonTitle}
                </Button>
            </div>
        );
    };

    const renderGridColumns = (): JSX.Element[] => {
        const columns: JSX.Element[] = [];
        const columnWidth = content ? '7' : '10';

        if (content) {
            columns.push(
                <GridColumn all="5" tablet="12" mobile="12" orderTablet="2" orderMobile="2" key={1}>
                    {renderContent(content)}
                </GridColumn>
            );
        }

        columns.push(
            <GridColumn all={columnWidth} tablet="12" mobile="12" key={0}>
                <div className={cn('video-wrapper')}>
                    <div className={cn('video-inner')}>
                        {renderSource(mediaType)}
                    </div>
                </div>
            </GridColumn>
        );

        return columns;
    };

    return (
        <div className={cn([className])}>
            <ContentArea outerBackgroundColor="white">
                <div className={cn('inner')}>
                    <Grid vAlign="center" hAlign="center">
                        {renderGridColumns()}
                    </Grid>
                </div>
            </ContentArea>
        </div>
    );
};

VideoBlock.propTypes = {
    content: PropTypes.shape({
        title: PropTypes.string.isRequired,
        description: PropTypes.array.isRequired,
        href: PropTypes.string,
        buttonTitle: PropTypes.string.isRequired,
        onButtonClick: PropTypes.func,
    }),
    mediaType: PropTypes.oneOf(['video', 'youtube', 'gif']),
    mediaSource: PropTypes.string.isRequired,
    className: PropTypes.string,
};

export default VideoBlock;
