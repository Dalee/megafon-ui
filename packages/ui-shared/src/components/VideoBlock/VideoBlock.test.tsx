import * as React from 'react';
import { shallow } from 'enzyme';
import VideoBlock, { Props } from './VideoBlock';
import { cnCreate } from '@megafon/ui-core';

const props: Props = {
    mediaSource: '',
    className: 'test-class-name',
};

const content = {
    title: 'Test title',
    description: [
        'Test description',
        'Test description',
    ],
    buttonTitle: 'Button title',
    href: '#',
    onButtonClick: jest.fn(),
};

const cn = cnCreate('mfui-video-block');

describe('<VideoBlock />', () => {
    it('it renders VideoBlock with default props', () => {
        const component = shallow(<VideoBlock mediaSource="">Test paragraph text</VideoBlock>);
        expect(component).toMatchSnapshot();
    });

    it('it renders VideoBlock with props', () => {
        const component = shallow(<VideoBlock {...props}>Test paragraph text</VideoBlock>);
        expect(component).toMatchSnapshot();
    });

    it('it renders VideoBlock with content', () => {
        const component = shallow(<VideoBlock {...props} content={content}>Test paragraph text</VideoBlock>);
        expect(component).toMatchSnapshot();
    });

    it('it renders VideoBlock with youtube media type', () => {
        const component = shallow(<VideoBlock {...props} mediaType="youtube">Test paragraph text</VideoBlock>);
        expect(component).toMatchSnapshot();
    });

    it('it renders VideoBlock with gif media type', () => {
        const component = shallow(<VideoBlock {...props} mediaType="gif">Test paragraph text</VideoBlock>);
        expect(component).toMatchSnapshot();
    });

    it('should call onClick props', () => {
        const component = shallow(<VideoBlock {...props} content={content}>Test paragraph text</VideoBlock>);
        const btn = component.find(`.${cn('button')}`);

        btn.simulate('click');
        expect(content.onButtonClick).toBeCalled();
    });
});
