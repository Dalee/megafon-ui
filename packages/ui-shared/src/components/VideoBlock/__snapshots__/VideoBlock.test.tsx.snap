// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoBlock /> it renders VideoBlock with content 1`] = `
<div
  className="mfui-beta-video-block"
>
  <Grid
    className="mfui-beta-video-block__grid"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="5"
      flex={false}
      grow={false}
      key="column-content"
      mobile="12"
      orderAll="0"
      orderDesktop="0"
      orderMobile="2"
      orderTablet="2"
      orderWide="0"
      tablet="12"
    >
      <div
        className="mfui-beta-video-block__content"
      >
        <Header
          as="h3"
          className="mfui-beta-video-block__header"
          color="black"
          hAlign="inherit"
        >
          Test title
        </Header>
        <div>
          <Paragraph
            className="mfui-beta-video-block__text"
            hasMargin={false}
            key="0Test description"
          >
            Test description
          </Paragraph>
          <Paragraph
            className="mfui-beta-video-block__text"
            hasMargin={false}
            key="1Test description2"
          >
            Test description2
          </Paragraph>
        </div>
        <Button
          className="mfui-beta-video-block__button"
          href="#"
        >
          Button title
        </Button>
      </div>
    </GridColumn>
    <GridColumn
      all="7"
      flex={false}
      grow={false}
      key="column-video"
      mobile="12"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="12"
    >
      <div
        className="mfui-beta-video-block__video-wrapper mfui-beta-video-block__video-wrapper_with-content"
      >
        <video
          autoPlay={true}
          className="mfui-beta-video-block__video"
          loop={true}
          muted={true}
        >
          <source
            src="video.mp4"
            type="video/mp4"
          />
        </video>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<VideoBlock /> it renders VideoBlock with default props 1`] = `
<div
  className="mfui-beta-video-block"
>
  <Grid
    className="mfui-beta-video-block__grid"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="10"
      flex={false}
      grow={false}
      key="column-video"
      mobile="12"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="12"
    >
      <div
        className="mfui-beta-video-block__video-wrapper"
      >
        <video
          autoPlay={true}
          className="mfui-beta-video-block__video"
          loop={true}
          muted={true}
        >
          <source
            src="video.mp4"
            type="video/mp4"
          />
        </video>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<VideoBlock /> it renders VideoBlock with sound turned on 1`] = `
<div
  className="mfui-beta-video-block"
>
  <Grid
    className="mfui-beta-video-block__grid"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="10"
      flex={false}
      grow={false}
      key="column-video"
      mobile="12"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="12"
    >
      <div
        className="mfui-beta-video-block__video-wrapper"
      >
        <video
          autoPlay={true}
          className="mfui-beta-video-block__video"
          loop={true}
          muted={false}
        >
          <source
            src="video"
            type="video/mp4"
          />
        </video>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<VideoBlock /> it renders VideoBlock with youtube media type 1`] = `
<div
  className="mfui-beta-video-block"
>
  <Grid
    className="mfui-beta-video-block__grid"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="10"
      flex={false}
      grow={false}
      key="column-video"
      mobile="12"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="12"
    >
      <div
        className="mfui-beta-video-block__video-wrapper"
      >
        <iframe
          allow="autoplay"
          frameBorder="0"
          height="100%"
          src="https://www.youtube.com/embed/youtube?&autoplay=1&mute=1&loop=1&rel=0&controls=0&showinfo=0e&iv_load_policy=3&playlist=youtube"
          width="100%"
        />
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`<VideoBlock /> it renders VideoBlock with youtube media type and sound turned on 1`] = `
<div
  className="mfui-beta-video-block"
>
  <Grid
    className="mfui-beta-video-block__grid"
    hAlign="center"
    multiRow={true}
  >
    <GridColumn
      all="10"
      flex={false}
      grow={false}
      key="column-video"
      mobile="12"
      orderAll="0"
      orderDesktop="0"
      orderMobile="0"
      orderTablet="0"
      orderWide="0"
      tablet="12"
    >
      <div
        className="mfui-beta-video-block__video-wrapper"
      >
        <iframe
          allow="autoplay"
          frameBorder="0"
          height="100%"
          src="https://www.youtube.com/embed/youtube?&autoplay=1&mute=0&loop=1&rel=0&controls=0&showinfo=0e&iv_load_policy=3&playlist=youtube"
          width="100%"
        />
      </div>
    </GridColumn>
  </Grid>
</div>
`;
